phpseclib/File/X509.php

```diff
@@ -4614,4 +4614,27 @@ function _extractBER($str)
         $temp = preg_match('#^[a-zA-Z\d/+]*={0,2}$#', $temp) ? base64_decode($temp) : false;
         return $temp != false ? $temp : $str;
     }
+
+    /**
+     * Returns the OID corresponding to a name
+     *
+     * What's returned in the associative array is either a name or an OID if no OID to name mapping is available.
+     * The problem with this is that what may be an unmapped OID in one version of phpseclib may not be unmapped
+     * in the next version, so apps that are looking at this OID may not be able to work from version to version.
+     *
+     * This method will return the OID if a name is passed to it and if no mapping is avialable it'll assume that
+     * what's being passed to it already is an OID and return that instead. A few examples.
```

Comment:
> I am not sure this is useful API. Why not return null or false?
The current behaviour can be easily kept by calling
`$x509->getOID($name) ?: $name`

Because `switch ($x509->getOID($name))` looks better than `switch ($x509->getOID($name) ?: $name)`? I figure most people using `$x509->getOID($name)` will be using it in switch statements, trying to make their switch statements future proof and with that as a use case most people are going to have to do `$x509->getOID($name) ?: $name` anyway. And if most people are going to have to do that outside of the function then why not do that inside of the function instead?

It's also not clear to me what `mapOID` would do? I guess that'd do something like...  `getOID('2.16.840.1.101.3.4.2.1') == 'id-sha256'`? If so I can add that.

So I guess lmk about `mapOID`. I'll give you a chance to further convince me about `getOID`. In the mean time I'll do the other changes.

Thanks!